# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1288315099 25200
#      Thu Oct 28 18:18:19 2010 -0700
# Node ID 5962c1cfdbcbedf478dd579f2aef1c95dacfc582
# Parent  254a87b40f60334aef18f3b24b8dd3b5bc3f150a
Deprecate autocor and autocov functions.

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2010-10-28  Rik  <octave@nomad.inbox5.com>
+
+	* NEWS: Update deprecated function list with autocov and autocor.
+
 2010-10-25  Kai Habel  <kai.habel@gmx.de>
 
 	* NEWS: Add uimenu.
 
 2010-10-24  Rik  <octave@nomad.inbox5.com>
 
 	* NEWS: Update deprecated function list for version 3.4.
 
diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -403,17 +403,17 @@ Summary of important user-visible change
       spchol              split
       spchol2inv          splu
 
     The following functions have been deprecated in Octave 3.4 and will
     be removed from Octave 3.8 (or whatever version is the second major
     release after 3.4):
 
       cellidx      fstat        values       gammai
-      betai        is_global
+      betai        is_global    autocor      autocov
 
 Summary of important user-visible changes for version 3.2:
 ---------------------------------------------------------
 
  ** Compatibility with Matlab graphics has been improved.
 
     The hggroup object and associated listener callback functions have
     been added allowing the inclusion of group objects.  Data sources
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-10-28  Rik  <octave@nomad.inbox5.com>
+
+	* deprecated/module.mk, signal/module.mk, deprecated/autocov.m,
+	  deprecated/autocor.m: Deprecate autocov and autocor functions.
+
 2010-10-27  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m: Don't attempt to plot patch outlines if
 	the marker property is set to none.
 
 2010-10-28  Ben Abbott <bpabbott@mac.com>
 
 	* plot/daspect.m, plot/pbaspect.m: Add demos setting both data and
diff --git a/scripts/signal/autocor.m b/scripts/deprecated/autocor.m
rename from scripts/signal/autocor.m
rename to scripts/deprecated/autocor.m
--- a/scripts/signal/autocor.m
+++ b/scripts/deprecated/autocor.m
@@ -18,23 +18,35 @@
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {} autocor (@var{x}, @var{h})
 ## Return the autocorrelations from lag 0 to @var{h} of vector @var{x}.
 ## If @var{h} is omitted, all autocorrelations are computed.
 ## If @var{x} is a matrix, the autocorrelations of each column are
 ## computed.
+## The particular algorithm used is from the field of statistics and 
+## differs from the definition used in signal processing.
 ## @end deftypefn
 
 ## Author: FL <Friedrich.Leisch@ci.tuwien.ac.at>
 ## Description: Compute autocorrelations
 
+## Deprecated in version 3.4
+
 function retval = autocor (X, h)
 
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "autocor is obsolete and will be removed from a future version of Octave; See the Octave-Forge signal package and the function xcor for a replacement");
+  endif
+
+
   if (nargin == 1)
     retval = autocov (X);
   elseif (nargin == 2)
     retval = autocov (X, h);
   else
     print_usage ();
   endif
 
diff --git a/scripts/signal/autocov.m b/scripts/deprecated/autocov.m
rename from scripts/signal/autocov.m
rename to scripts/deprecated/autocov.m
--- a/scripts/signal/autocov.m
+++ b/scripts/deprecated/autocov.m
@@ -18,22 +18,32 @@
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {} autocov (@var{x}, @var{h})
 ## Return the autocovariances from lag 0 to @var{h} of vector @var{x}.
 ## If @var{h} is omitted, all autocovariances are computed.
 ## If @var{x} is a matrix, the autocovariances of each column are
 ## computed.
+## The particular algorithm used is from the field of statistics and 
+## differs from the definition used in signal processing.
 ## @end deftypefn
 
 ## Author: FL <Friedrich.Leisch@ci.tuwien.ac.at>
 ## Description: Compute autocovariances
 
+## Deprecated in version 3.4
+
 function retval = autocov (X, h)
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "autocov is obsolete and will be removed from a future version of Octave; See the Octave-Forge signal package and the function xcov for a replacement");
+  endif
 
   [n, c] = size (X);
 
   if (isvector (X))
     n = length (X);
     c = 1;
     X = reshape (X, n, 1);
   endif
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -1,11 +1,13 @@
 FCN_FILE_DIRS += deprecated
 
 deprecated_FCN_FILES = \
+  deprecated/autocor.m \
+  deprecated/autocov.m \
   deprecated/betai.m \
   deprecated/cellidx.m \
   deprecated/clg.m \
   deprecated/complement.m \
   deprecated/create_set.m \
   deprecated/dmult.m \
   deprecated/fstat.m \
   deprecated/gammai.m \
diff --git a/scripts/signal/module.mk b/scripts/signal/module.mk
--- a/scripts/signal/module.mk
+++ b/scripts/signal/module.mk
@@ -1,17 +1,15 @@
 FCN_FILE_DIRS += signal
 
 signal_FCN_FILES = \
   signal/arch_fit.m \
   signal/arch_rnd.m \
   signal/arch_test.m \
   signal/arma_rnd.m \
-  signal/autocor.m \
-  signal/autocov.m \
   signal/autoreg_matrix.m \
   signal/bartlett.m \
   signal/blackman.m \
   signal/detrend.m \
   signal/diffpara.m \
   signal/durbinlevinson.m \
   signal/fftconv.m \
   signal/fftfilt.m \
